src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,378 @@
+function matroid_realization_space(M::Matroid, A::Vector{Int}, F::AbstractAlgebra.Ring=ZZ)
+    d = rank(M)
+    n = length(matroid_groundset(M))
+
+    Bs = bases(M)
+    R, x, xdict = realization_polynomial_ring(Bs,A,F)
+    return matroid_realization_space_given_ring(d, n, M, F, A, R, x, xdict)
+end
+
+
+# given the bases Bs of a matroid, and a fixed basis B, this function finds
+# the nonzero coordinates xij of the coordinate ring of the matroid stratum,
+# These correspond to all elements A of Bs such that the symmetric difference
+# with B has exactly 2 elements. 
+
+function bases_matrix_coordinates(Bs::Vector{Vector{Int}}, B::Vector{Int})
+    
+    coord_bases = [b for b in Bs if length(symdiff(B,b)) == 2]
+    
+    new_coords = Vector{Vector{Int}}([])
+    
+    for b in coord_bases
+        row_b = setdiff(B,b)[1]
+        row_b = count(a->(a<row_b), B) + 1
+        # count(f, v) does what?
+        # - iterate through the elements a in v
+        # - compute f(a) 
+        # - if that is true, increment the counting variable by 1
+        # - otherwise, continue
+        # - return the value of the internal counter.
+        # Similar with all(a->(a<row_b), B), for instance.
+        #row_b = length([a for a in B if a < row_b]) + 1
+        
+        col_b = setdiff(b,B)[1]
+        col_b = col_b - length([a for a in B if a ≤ col_b]) 
+        
+        push!(new_coords, [row_b,col_b])
+    end
+    return sort!(new_coords, by = x -> (x[2], x[1]))    
+end
+
+
+
+# Given the bases Bs of a matroid, a fixed basis B, and a coefficient field F
+# this function creates a polynomial ring in xij, where the xij are 
+# determined by the function basis_matrix_coordinates. This function also
+# returns (as the 3rd element of a triple) a dictionary (i,j) => xij.
+
+function make_polynomial_ring(Bs::Vector{Vector{Int}}, B::Vector{Int},
+                              F::AbstractAlgebra.Ring)
+    
+    MC = bases_matrix_coordinates(Bs, B)
+    R, x = PolynomialRing(F, :"x"=>MC)
+    xdict = Dict{Vector{Int}, MPolyElem}([MC[i] => x[i] for i in 1:length(MC)])
+    return R, x, xdict
+end
+
+
+# This function returns a d x (n-d) matrix with values in the polynomial ring
+# created by make_polynomial_ring. The entries are xij, except where the
+# value is 0, as determined by the nonbases. 
+
+function make_coordinate_matrix_no_identity(d::Int, n::Int,
+                                            MC::Vector{Vector{Int}},
+                                            R::MPolyRing, x::Vector{T},
+                                            xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    S = MatrixSpace(R, d, n-d)
+    X = S()
+    for j in 1:n-d, i in 1:d
+        if [i,j] in MC
+            X[i,j] = xdict[[i,j]]
+        else
+            X[i,j] = R(0)
+        end
+    end
+    return X
+end
+
+
+# M and N have same number of rows, and M has #B columns, both have entries in ring R
+function interlace_columns(M, N, B::Vector{Int},
```

Comment:
```suggestion
function interlace_columns(M::MatrixElem, N::MatrixElem, B::Vector{Int},
```
It seems that `M` and `N` are matrices. But I'm not sure, these annotations are actually correct! Do you use Julia or Oscar matrices here?

Reply:
I will try your suggestion, I think it is correct, at least this is what worked in the bases_determinants function.